staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go

```diff
@@ -132,68 +131,67 @@ func buildPropertiesFromGrpc(device *dmiapi.Device) []common.DeviceProperty {
 		var dbconfig common.DBConfig
 		var pushMethod []byte
 		var pushMethodName string
-		if pptv.PushMethod != nil {
-			if pptv.PushMethod.DBMethod != nil {
-				dbMethodName, err = getDBMethodFromGrpc(pptv)
+		if pptv.PushMethod != nil && pptv.PushMethod.DBMethod != nil {
+			dbMethodName, err = getDBMethodFromGrpc(pptv)
```

Comment:
In order to fix the null pointer problem before, we put pptv.PushMethod != nil in a separate if. If you change it like this, if the user does not configure the http or mqtt field of pushMethod, it seems that the null pointer problem will reappear on line 178.

Reply:
What I mean is that after this if, the pushMethodName field will be parsed on line 175. pptv.PushMethod != nil is not judged on line 175. This part of the original code is in the if of pptv.PushMethod != nil.

Reply:
![pushmethod](https://github.com/kubeedge/kubeedge/assets/48212997/555ca9ae-57fa-42d6-9b99-cbef82183ee4)

Reply:
> What I mean is that after this if, the pushMethodName field will be parsed on line 175. pptv.PushMethod != nil is not judged on line 175. This part of the original code is in the if of pptv.PushMethod != nil.

OK, I get it.

Reply:
Oh, yes. But if that's the case, these `if` is confusing, so let me fix it again